repose-aggregator/components/filters/valkyrie-authorization/src/test/scala/org/openrepose/filters/valkyrieauthorization/ValkyrieAuthorizationFilterTest.scala

```diff
@@ -427,6 +427,65 @@ class ValkyrieAuthorizationFilterTest extends FunSpec with BeforeAndAfter with M
     }
   }
 
+  describe("when permission to role translation is turned on and request includes a Device ID") {
+    val config = createGenericValkyrieConfiguration(null)
+    config.setTranslatePermissionsToRoles(new Object)
+    val tenantId = "hybrid:98765"
+    val transformedTenant = "98765"
+    val contactId = "123456"
+    val deviceId = "98765"
+    val filterChain = mock[FilterChain]
+    val mockServletRequest = new MockHttpServletRequest
+    val mockServletResponse = new MockHttpServletResponse
+    val filter = new ValkyrieAuthorizationFilter(mock[ConfigurationService], akkaServiceClient, mockDatastoreService)
+    filter.configurationUpdated(config)
+
+    def setup() = {
+      mockServletRequest.resetAll()
+      mockServletRequest.setMethod("GET")
+      mockServletRequest.setRequestURL("http://foo.com:8080")
+      mockServletRequest.setHeader("X-Tenant-Id", tenantId)
+      mockServletRequest.setHeader("X-Contact-Id", contactId)
+
+      mockServletResponse.resetAll()
+
+      Mockito.reset(filterChain)
+    }
+
+    List("view_product", "edit_product", "admin_product").foreach { devicePermission =>
+      it(s"should translate permissions to roles with device permission $devicePermission") {
+        setup()
```

Comment:
Not a huge fan of calling this for every test, but... Scalatest doesn't seem to have an easy/simple way of running some code before some tests.

Apparently you can write multiple test suites and combine them, but depending on the test runner being used, that could cause tests to run more than once.

Ugh.